src/codegen/codegen_ispc_visitor.cpp

```diff
@@ -477,12 +477,17 @@ void CodegenIspcVisitor::print_headers_include() {
 }
 
 void CodegenIspcVisitor::print_nmodl_constants() {
-    printer->add_newline(2);
-    printer->add_line("/** constants used in nmodl. */");
-    // we use here macros to work around ispc's PI being declared in global namespace
-    printer->add_line("static const uniform double FARADAY = 96485.3d;");
-    printer->add_line("static const uniform double ISPC_PI = 3.14159d;");
-    printer->add_line("static const uniform double R = 8.3145d;");
+    if (!info.factor_definitions.empty()) {
+        printer->add_newline(2);
+        printer->add_line("/** constants used in nmodl */");
+        for (auto& it: info.factor_definitions) {
+            const std::string name = it->get_node_name() == "PI" ? "ISPC_PI" : it->get_node_name();
+            std::string format_string = "static const uniform double {} = {};";
+            printer->add_line(fmt::format(format_string.c_str(),
+                                          name,
+                                          double_to_string(it->get_value()->get_value())));
```

Comment:
`double_to_string` is printing all the doubles with 6 decimals, since it's using the `%f` to print them, while `mod2c` is using 6 or 18 digits depending on if we have legacy or non legacy units.
I think we should edit `double_to_string` or use a different function to have the same behaviour in this case.
Maybe this is not needed though after @pramodk 's change were all the doubles will be printed as strings.

Reply:
Discussed this offline - we will fix this in next PR.